include/seqan3/core/metafunction/basic.hpp

```diff
@@ -0,0 +1,68 @@
+// ============================================================================
```

Comment:
Can we put this file in `/seqan3/core/type_traits.hpp`?

Because we redefine stuff from the stl and we should keep the naming convention?

Reply:
you mean rename the folder? We could discuss this, but maybe after the retreat?

Reply:
? No, we should should rename the file `basic.hpp` into `type_traits.hpp`